test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java

```diff
@@ -81,15 +81,16 @@ public boolean clusterHasFeature(String featureId) {
         Matcher matcher = VERSION_FEATURE_PATTERN.matcher(featureId);
         if (matcher.matches()) {
             Version extractedVersion = Version.fromString(matcher.group(1));
-            if (Version.V_8_14_0.before(extractedVersion)) {
+            if (Version.fromString("8.15.0").before(extractedVersion)) {
```

Comment:
Referencing versions released in this branch's 'future' is not great, can you give some background to this PR?

Reply:
Reason for this is this backport: #109448

Though, actually, the YAML spec should be modified to use 8.14.1 I suppose, rather than 8.15.0.
This could be `Version.CURRENT` in that case.

Reply:
I think that would be preferable, else we'll be referencing versions that dont exist yet.

The `gte_` conditions on 8.14 and main should be updated to be `gte_v8.14.1` after this though

Reply:
I think that would be preferable, else we'll be referencing versions that dont exist yet.

The `gte_` conditions on 8.14 and main should be updated to be `gte_v8.14.1` after this though